pkg/controller/plan/adapter/ovirt/client.go

```diff
@@ -64,16 +65,44 @@ func (r *Client) RemoveSnapshots(vmRef ref.Ref, precopies []planapi.Precopy) (er
 		return
 	}
 	snapsService := vmService.SnapshotsService()
-	for i := range precopies {
-		snapService := snapsService.SnapshotService(precopies[i].Snapshot)
-		_, err = snapService.Remove().Async(true).Query("correlation_id", r.Migration.Name).Send()
-		if err != nil {
-			err = liberr.Wrap(err)
+	go func() {
+		for i := range precopies {
+			snapService := snapsService.SnapshotService(precopies[i].Snapshot)
+			correlationID := fmt.Sprintf("precopy_%d_finalize", i)
+			_, err = snapService.Remove().Query("correlation_id", correlationID).Send()
```

Comment:
how about trying several times to initiate the first remove-snapshot when it fails (due to memory lock) as we know it would most probably happen due to image transfer

Reply:
You mean checking if it fails generally or fails because of the memory lock? Do we have a reliable way to know it failed because because of a memory lock? Because I'm only aware of parsing the error message which is probably not as reliable

Reply:
we talked about this offline that we don't have a good/reliable way to distinguish between failures to acquire the lock and other failures, but in this particular case it seems reasonable to assume that failure to initiate the snapshot removals are due to image transfer and try several time, unless we have a better way to sync with what's going on in CDI

Reply:
Thought about this some more, image transfer entities have an image id (which is actually the disk id)
```xml
<image_transfer href="/ovirt-engine/api/imagetransfers/e2294cfc-9b41-43eb-b7ef-fcffd15ce9b9" id="e2294cfc-9b41-43eb-b7ef-fcffd15ce9b9">
	  <actions>
	  ...
	  </actions>
	  <active>true</active>
	  <direction>download</direction>
	  <format>raw</format>
	  <inactivity_timeout>60</inactivity_timeout>
	  <phase>finished_success</phase>
	  <shallow>false</shallow>
	  <timeout_policy>cancel</timeout_policy>
	  <transferred>1427243008</transferred>
	  <host href="/ovirt-engine/api/hosts/00509b81-fa25-4953-ba07-824dfc4e32d0" id="00509b81-fa25-4953-ba07-824dfc4e32d0"/>
	  <image id="790bd328-0525-45c5-925e-52f3b4428414"/>
</image_transfer>
``

So I I'll see if simply searching if the disk id is participating in any active transfer is enough

Reply:
we can do something like optimistic locking :) trying to remove it and if that fails confirming that it's due to an ongoing image transfer this way